### PR TITLE
feat: base64 decoder for cw3 wasm messages

### LINKED ISSUE #1725: feat: base64 decoder for cw3 wasm messages
A stargate message has the form

```json
{
  "stargate": {
    "type_url": "/cosmos.staking.v1beta1.MsgUndelegate",
    "value": "Cj9uaWJpMTdwOXJ6d25uZnhjanAzMnVuOXVnN3loaHpndGtodmw5amZrc3p0Z3c1dWg2OXdhYzJwZ3N5bjcwbmoSMm5pYml2YWxvcGVyMXdqNWtma25qa3BjNmpkMzByeHRtOHRweGZqZjd4cWx3eDM4YzdwGgwKBXVuaWJpEgMxMTE="
  }
}
```

so it's very difficult to tell exactly what the message is doing. 

Voters of a cw3/cw4 multisig can't make an informed decision without a tool to decode the base64 bytes.

### PR DESCRIPTION
**Pull request by matthiasmatt**
# Purpose / Abstract

- Closes #1725 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new command-line functionality for decoding base64-encoded protobuf messages.

- **Tests**
	- Implemented tests to ensure the reliability and correctness of the new decoding feature.

- **Documentation**
	- Updated command-line interface documentation to include the new decoding command.

### CODE REVIEW

File: cmd/nibid/cmd/decode_base64.go [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+		"/nibiru.spot.v1.MsgSwapAssets":                               new(spottypes.MsgSwapAssets),
+		"/nibiru.sudo.v1.MsgChangeRoot":                               new(sudotypes.MsgChangeRoot),
+		"/nibiru.sudo.v1.MsgEditSudoers":                              new(sudotypes.MsgEditSudoers),
+		"/nibiru.tokenfactory.v1.MsgBurn":                             new(tokenfactorytypes.MsgBurn),
+		"/nibiru.tokenfactory.v1.MsgChangeAdmin":                      new(tokenfactorytypes.MsgChangeAdmin),
+		"/nibiru.tokenfactory.v1.MsgCreateDenom":                      new(tokenfactorytypes.MsgCreateDenom),
+		"/nibiru.tokenfactory.v1.MsgMint":                             new(tokenfactorytypes.MsgMint),
+		"/nibiru.tokenfactory.v1.MsgSetDenomMetadata":                 new(tokenfactorytypes.MsgSetDenomMetadata),
+		"/nibiru.tokenfactory.v1.MsgUpdateModuleParams":               new(tokenfactorytypes.MsgUpdateModuleParams),
+	}
```
[the remark this answers is not in the document]
Comment (matthiasmatt) [reply cmd/nibid/cmd/decode_base64.go:189]: @Unique-Divine told you the bot would be spitting in my face

[the remark this answers is not in the document]
Comment (matthiasmatt) [reply cmd/nibid/cmd/decode_base64.go:189]: @coderabbitai do you know how?

File: cmd/nibid/cmd/decode_base64.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	proto "github.com/cosmos/gogoproto/proto"
+	"github.com/spf13/cobra"
+
+	// stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
```
Comment (k-yang) [inline cmd/nibid/cmd/decode_base64.go:12]:
```suggestion
```

File: cmd/nibid/cmd/decode_base64.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (k-yang) [reply cmd/nibid/cmd/decode_base64.go:189]: Yes, please explain the dynamic iteration method further.

[the remark this answers is not in the document]
Comment (matthiasmatt) [reply cmd/nibid/cmd/decode_base64.go:188]: Nevermind, found that InterfaceRegistry.Resolve works when interface is properly initilized (which it wasn't in our test cases)

File: cmd/nibid/cmd/decode_base64.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	cmd := &cobra.Command{
+		Use:   "base64-decode",
+		Short: "Decode a base64-encoded protobuf message",
+		Long: `Decode a base64-encoded protobuf message from JSON input.
+The input should be a JSON object with 'type_url' and 'value' fields.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+
+			outMessage, err := DecodeBase64StargateMsgs([]byte(args[0]), clientCtx)
+			fmt.Println(outMessage)
```
Comment (Unique-Divine) [inline cmd/nibid/cmd/decode_base64.go:117]: Would put if err == nil, print so that it only runs on success

**Review by Unique-Divine — APPROVED** [review]

**Pull request merged**